client-side/src/components/settings/Preferences.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState,useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next'
+import Select from '../../stories/Select/Select.jsx'
+import CONSTANTS from './constantSetting.js'
+import moment from 'moment-timezone';
+import './Preferences.scss';
+
+const createTimeZones = () => {
+  return moment.tz.names().map(timezone => ({
+    value: timezone,
+    text: timezone,
+  }));
+};
+
+const Preferences = ({ currentUser, onUpdate}) => {
+  const { t, i18n } = useTranslation();
+  const {  LABELS, LANGUAGE,DATE_FORMATS } = CONSTANTS;
+  const { timeZone: initialTimeZone, dateFormat: initialDateFormat } = currentUser.preference;
+  const [language, setLanguage] = useState(i18n.resolvedLanguage);
+  const [timeZone, setTimeZone] = useState(initialTimeZone);
+  const [dateFormat, setDateFormat] = useState(initialDateFormat)
+
+  useEffect(() => {
+    onUpdate({
+      language,
+      timeZone,
+      dateFormat
+    });
+  }, [language, timeZone, dateFormat, onUpdate]);
+
+  const handleLanguageChange = (value) => {
+    i18n.changeLanguage(value);
+    setLanguage(value)
+  }
+
+  const handleChangeTimeZone = (selectedTimeZone) => {
+    setTimeZone(selectedTimeZone);
+  };
+
+  return (
+    <div>
+      <Select
+        title={t(LABELS.SELECT_LANGUAGES)}
+        options={Object.keys(LANGUAGE).map(key => ({
```

Comment:
please move the map functionality to a variable and use the result here - please do it in all your code

Reply:
done